odoo/addons/base/models/res_users.py

```diff
@@ -714,6 +716,7 @@ def _login(cls, db, login, password, user_agent_env):
         if not password:
             raise AccessDenied()
         ip = request.httprequest.environ['REMOTE_ADDR'] if request else 'n/a'
+        user_agent = user_agent_env.get('HTTP_USER_AGENT')
```

Comment:
Is it really necessary to add the user-agent to the environment? Cannot we do like for the IP address?
```suggestion
        user_agent = request.httprequest.environ['HTTP_USER_AGENT'] if request else 'n/a'  
```

Reply:
the code inside `http.py` is still present